modules/internal_fns.py

```diff
@@ -183,37 +185,44 @@ def obs_array(dates_obs, lat_idx, lon_idx):
         tmp_error_storage = []
 
         for i, ncfile in enumerate(files):
-
-            data_temp = nc.Dataset(ncfile)
-            nc_value = data_temp.variables[obs_var][:, lat_idx, lon_idx]
-            # Check if masked
-            # TODO: Check if there is a better way to do this
-            if np.ma.is_masked(nc_value):
-                nc_value = nc_value.filled(np.nan)
-            else:
-                nc_value = np.ma.getdata(nc_value)
-
-            tmp_obs_storage.extend(nc_value)
-
-            # do the same conditionally for errors
-
-            if r_cov == 'dynamic_error':
-                nc_value = data_temp.variables[cfg.obs_error_var_names[cont]
-                                               ][:, lat_idx, lon_idx]
+            
+            try:
+            
+                data_temp = nc.Dataset(ncfile)
+                nc_value = data_temp.variables[obs_var][:, lat_idx, lon_idx]
                 # Check if masked
                 # TODO: Check if there is a better way to do this
                 if np.ma.is_masked(nc_value):
                     nc_value = nc_value.filled(np.nan)
                 else:
                     nc_value = np.ma.getdata(nc_value)
-
-                tmp_error_storage.extend(nc_value)
-            else:
-
-                tmp_error_storage = [r_cov[cont]] * len(tmp_obs_storage)
-
+    
+                tmp_obs_storage.extend(nc_value)
+            
+
+                # do the same conditionally for errors
+    
+                if r_cov == 'dynamic_error':
+                    
+                    nc_value = data_temp.variables[cfg.obs_error_var_names[cont]
+                                                   ][:, lat_idx, lon_idx]
+                    # Check if masked
+                    # TODO: Check if there is a better way to do this
+                    if np.ma.is_masked(nc_value):
+                        nc_value = nc_value.filled(np.nan)
+                    else:
+                        nc_value = np.ma.getdata(nc_value)
+    
+                    tmp_error_storage.extend(nc_value)
+                else:
+    
+                    tmp_error_storage = [r_cov[cont]] * len(tmp_obs_storage)
+            except:
```

Comment:
please do not use bare excepts. Add the exceptions you expect to find. If there was a problem in a corrupt netcdf for example, we want MuSA to throw an exception.